src/certgen_certificate_generator_server.cc

```diff
@@ -27,11 +27,285 @@
+
+                    auto it = std::find (result.begin(), result.end(), host);
+                    if (it == result.end())
+                    {
+                        result.push_back (host);
+                    }
+                }
+            }
+        }
+        freeifaddrs (interfaces);
+        return result;
+    }
+
+
+    std::string CertificateGeneratorServer::handleGenerateSelfsignedCertificate(const fty::Payload & params)
+    {
+        // load config file for the service
+        CertificateGeneratorConfig certgenConfig;
+
+        if (params.empty() || params[0].empty ())
+        {
+            throw std::runtime_error ("Missing service name");
+        }
+
+        std::string serviceName (params[0]);
+        std::string configFilePath(CONFIG_PATH + serviceName + ".cfg");
+        std::ifstream configFile (configFilePath);
+        if (!configFile)
+        {
+            throw std::runtime_error ("Could not open file " + configFilePath);
+        }
+
+        configFile.exceptions ( std::ifstream::failbit | std::ifstream::badbit );
+        std::stringstream configJson;
+        configJson << configFile.rdbuf();
+        configFile.close ();
+
+        cxxtools::SerializationInfo certgenSi;
+        cxxtools::JsonDeserializer deserializer (configJson);
+        deserializer.deserialize (certgenSi);
+
+        certgenSi >>= certgenConfig;
+
+        fty::CertificateConfig config;
+        // fill the configuration data
+        uint8_t version;
+        std::istringstream versionStr (certgenConfig.version());
+        versionStr >> version;
+        config.setVersion(version);
+
+        std::chrono::hours validityOffset (HOURS_IN_DAY * certgenConfig.certConf().getValidityOffset());
+        std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds> currentTimestamp =
+            std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now());
+        std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds> startTimestamp = currentTimestamp + validityOffset;
+        config.setValidFrom(startTimestamp.time_since_epoch().count());
+
+        std::chrono::hours validity (HOURS_IN_DAY * certgenConfig.certConf().getValidity());
+        currentTimestamp = std::chrono::time_point_cast<std::chrono::seconds>(std::chrono::system_clock::now());
+        std::chrono::time_point<std::chrono::system_clock, std::chrono::seconds> endTimestamp = currentTimestamp + validity;
+        config.setValidTo(endTimestamp.time_since_epoch().count());
+
+        config.setCountry(COUNTRY);
+        config.setState(STATE);
+        config.setLocality(LOCALITY);
+        config.setOrganization(ORGANIZATION);
+        config.setOrganizationUnit(ORGANIZATION_UNIT);
+        config.setCommonName(COMMON_NAME);
+        config.setEmail(EMAIL);
+
+        std::list<std::string> ipList = getSystemIPs (AF_INET);
+        ipList.merge (getSystemIPs (AF_INET6));
+        config.setIpList(ipList);
```

Comment:
IPs need to be set if we are in static mode only. In dhcp we need only FQDN (as the IP will change overtime)

Reply:
Addressed.